src/app/nonprofit/nonprofit-page/nonprofit-page.component.ts

```diff
@@ -284,7 +297,75 @@ export class DonationDialogComponent {
 
   submitDonation(): void
   {
-    // Clearing variables
+    // table name
+    const table = 'Donations';
+
+    const usersDonation = {
+      orgName: this.getOrgName(),
+      orgEIN: this.getNonProfitEIN(),
+      donationDate: new Date(),
+      donationAmount: this.getDonationAmount(),
+      cardHolderName: this.getCardHolderName(),
+      cardHolderStreet: this.getStreetAddress(),
+      cardHolderState: this.getState(),
+      cardHolderEmail: this.getEmail()
+    };
+
+    // Get the user's ID
+    this.authService.getUserId().then((userUID) => {
+
+      // Get user's past donations
+      this.firestoreDB.collection(table).doc(userUID).get()
+      .subscribe((snapshot: any) => {
+
+        // Check to see if user exists on table, if user does not exist
+        // then user has never made donations.
+        if(snapshot.data() != undefined)
```

Comment:
Instead of doing a read, loading in, and loading out, use firebase's `.set(..., {merge: true})`, along with `arrayUnion` method. an example is shown in the favorites array or ratings array in the nonprofit page component. this will create (or append) to an array in firebase. we can leave it as is for the sake of getting this merged in, or you can make the change if you want. it would probably just be 30 minute time requirement